github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8138730?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8138730?buildTab=artifacts#meta) on refs/heads/master @ [811a8c0e741b](https://github.com/cockroachdb/pebble/commits/811a8c0e741b):


```
=== CONT  TestMeta/execution/random-016
=== RUN   TestMeta/execution/standard-022
=== PAUSE TestMeta/execution/standard-022
=== CONT  TestMeta/execution/standard-022
=== RUN   TestMeta/compare/random-001
=== RUN   TestMeta/compare/standard-003
=== RUN   TestMeta/compare/standard-007
=== RUN   TestMeta/compare/standard-023
=== RUN   TestMeta/execution/standard-008
=== PAUSE TestMeta/execution/standard-008
=== CONT  TestMeta/execution/standard-008
=== RUN   TestMeta/execution/standard-024
=== PAUSE TestMeta/execution/standard-024
=== CONT  TestMeta/execution/standard-024
=== RUN   TestMeta/compare/standard-001
=== RUN   TestMeta/compare/standard-008
=== RUN   TestMeta/compare/standard-011
=== RUN   TestMeta/execution/random-019
=== PAUSE TestMeta/execution/random-019
=== CONT  TestMeta/execution/random-019
=== RUN   TestMeta/execution/random-021
=== PAUSE TestMeta/execution/random-021
=== CONT  TestMeta/execution/random-021
=== RUN   TestMeta/execution/standard-023
=== PAUSE TestMeta/execution/standard-023
=== CONT  TestMeta/execution/standard-023
=== RUN   TestMeta/compare/random-006
=== RUN   TestMeta/compare/standard-010
=== RUN   TestMeta/compare/standard-019
=== RUN   TestMeta/execution/random-003
=== PAUSE TestMeta/execution/random-003
=== CONT  TestMeta/execution/random-003
=== RUN   TestMeta/execution/standard-002
=== PAUSE TestMeta/execution/standard-002
=== CONT  TestMeta/execution/standard-002
=== RUN   TestMeta/execution/random-025
=== PAUSE TestMeta/execution/random-025
=== CONT  TestMeta/execution/random-025
=== RUN   TestMeta/execution/standard-012
=== PAUSE TestMeta/execution/standard-012
=== CONT  TestMeta/execution/standard-012
=== RUN   TestMeta/compare/random-008
=== RUN   TestMeta/compare/standard-004
=== RUN   TestMeta/compare/standard-005
=== RUN   TestMeta/execution/random-020
=== PAUSE TestMeta/execution/random-020
=== CONT  TestMeta/execution/random-020
=== RUN   TestMeta/execution/random-023
=== PAUSE TestMeta/execution/random-023
=== CONT  TestMeta/execution/random-023
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1672389719935244295 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/8208178?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/8208178?buildTab=artifacts#meta) on refs/heads/master @ [bc6b7180a19f](https://github.com/cockroachdb/pebble/commits/bc6b7180a19f):


```
        // INFO: [JOB 95] MANIFEST deleted 000419
        // INFO: [JOB 106] flushed 1 memtable to L0 [000461 000462 000464 000465 000467 000469 000470 000472 000473 000474 000475 000477 000478 000482 000483 000484 000485] (15 K), in 0.1s (0.3s total), output rate 105 K/s
        // INFO: [JOB 106] WAL deleted 000421
        // INFO: [JOB 110] compacting(default) L5 [000456] (2.0 K) + L6 [000226] (1020 B)
        // INFO: [JOB 111] compacting(default) L0 [000340 000341 000342 000386 000385 000384 000437 000438 000478] (9.2 K) + L5 [000480] (1.8 K)
        // INFO: [JOB 111] compacting: sstable created 000487
        // INFO: [JOB 110] compacting: sstable created 000489
        // INFO: [JOB 110] compacting: sstable created 000490
        // INFO: [JOB 111] compacting: sstable created 000491
        // INFO: [JOB 112] WAL created 000488
        // INFO: write stall ending
        // INFO: [JOB 114] flushing 1 memtable to L0
        // INFO: [JOB 114] flushing: sstable created 000492
        // INFO: [JOB 111] compacting: sstable created 000494
        // INFO: [JOB 110] compacted(default) L5 [000456] (2.0 K) + L6 [000226] (1020 B) -> L6 [000489 000490] (2.3 K), in 0.0s (0.0s total), output rate 184 K/s
        // INFO: [JOB 114] flushing: sstable created 000493
        // INFO: [JOB 111] compacting: sstable created 000495
        // INFO: [JOB 114] flushing: sstable created 000496
        // INFO: [JOB 114] flushing: sstable created 000497
        // INFO: [JOB 114] flushing: sstable created 000498
        // INFO: [JOB 111] compacting: sstable created 000499
        // INFO: [JOB 116] compacting(move) L5 [000458] (1.6 K) + L6 [] (0 B)
        // INFO: [JOB 116] compacted(move) L5 [000458] (1.6 K) + L6 [] (0 B) -> L6 [000458] (1.6 K), in 0.0s (0.0s total), output rate 124 M/s
        // INFO: [JOB 114] flushing: sstable created 000500
        // INFO: [JOB 114] flushing: sstable created 000501
        // INFO: [JOB 111] compacted(default) L0 [000340 000341 000342 000386 000385 000384 000437 000438 000478] (9.2 K) + L5 [000480] (1.8 K) -> L5 [000487 000491 000494 000495 000499] (6.7 K), in 0.2s (0.2s total), output rate 38 K/s
        // INFO: [JOB 111] sstable deleted 000478
        // INFO: [JOB 111] sstable deleted 000480
        // INFO: [JOB 114] flushing: sstable created 000502
        // INFO: [JOB 118] compacting(default) L5 [000491] (1.2 K) + L6 [000358] (998 B)
        // INFO: [JOB 118] compacting: sstable created 000503
        // INFO: [JOB 118] compacted(default) L5 [000491] (1.2 K) + L6 [000358] (998 B) -> L6 [000503] (1.1 K), in 0.0s (0.0s total), output rate 538 K/s
        // INFO: [JOB 110] sstable deleted 000456
        // INFO: [JOB 120] compacting(move) L5 [000487] (1.6 K) + L6 [] (0 B)
        // INFO: [JOB 118] sstable deleted 000491
        // INFO: [JOB 121] compacting(default) L5 [000499] (1.2 K) + L6 [000350] (1.1 K)
        // INFO: [JOB 114] MANIFEST created 000504
        // INFO: [JOB 121] compacting: sstable created 000505
        // INFO: [JOB 114] flushed 1 memtable to L0 [000492 000493 000496 000497 000498 000500 000501 000502] (6.7 K), in 0.3s (0.4s total), output rate 20 K/s
        // INFO: [JOB 120] compacted(move) L5 [000487] (1.6 K) + L6 [] (0 B) -> L6 [000487] (1.6 K), in 0.0s (0.1s total), output rate 47 M/s
        // INFO: [JOB 114] WAL deleted 000459
        // INFO: [JOB 114] MANIFEST deleted 000444
        // INFO: [JOB 122] compacting(default) L0 [000289 000291 000347 000392 000443 000484] (4.9 K) + L5 [] (0 B)
        // INFO: [JOB 121] compacted(default) L5 [000499] (1.2 K) + L6 [000350] (1.1 K) -> L6 [000505] (1.2 K), in 0.1s (0.1s total), output rate 11 K/s
        // INFO: [JOB 124] compacting(default) L5 [000451] (1.0 K) + L6 [000403] (1.1 K)
        // INFO: [JOB 124] compacting: sstable created 000506
        // INFO: [JOB 124] compacted(default) L5 [000451] (1.0 K) + L6 [000403] (1.1 K) -> L6 [000506] (1.2 K), in 0.0s (0.0s total), output rate 560 K/s
        // INFO: [JOB 109] ingested L0:000486 (1.0 K)
        db.Ingest(batch14) // <nil> #1898
        --- FAIL: TestMeta/execution/random-008 (8.73s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1672991349775368569 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8223711?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8223711?buildTab=artifacts#meta) on refs/heads/master @ [65ff304d3d7a](https://github.com/cockroachdb/pebble/commits/65ff304d3d7a):


```
        // INFO: [JOB 1008] compacting(default) L0 [001913 002019 002020 002018 002063] (5.5 K) + L4 [002087 001987 002035] (4.0 K)
        // INFO: [JOB 1003] sstable deleted 001910
        // INFO: [JOB 1003] sstable deleted 001911
        // INFO: [JOB 1003] sstable deleted 001986
        // INFO: [JOB 1003] sstable deleted 002017
        // INFO: [JOB 1003] sstable deleted 002081
        // INFO: [JOB 1010] compacting(default) L4 [002043] (990 B) + L5 [] (0 B)
        // INFO: [JOB 1010] compacting: sstable created 002089
        db.Merge("bydm@10", "c") // <nil> #7891
        db.Get("duvfraklhg@17") // [""] pebble: not found #7892
        // INFO: [JOB 1010] compacted(default) L4 [002043] (990 B) + L5 [] (0 B) -> L5 [002089] (990 B), in 0.0s (0.0s total), output rate 9.4 M/s
        db.Set("wcobceorrx@8", "iobmhgqvwymqfy") // <nil> #7893
        // INFO: [JOB 1012] compacting(delete-only) L5 [001604 002080] (1.7 K)
        // INFO: [JOB 1012] compacted(delete-only) L5 [001604 002080] (1.7 K) -> L6 [] (0 B), in 0.0s (0.0s total), output rate 0 B/s
        // INFO: [JOB 1013] compacting(default) L4 [001891] (1.0 K) + L5 [] (0 B)
        // INFO: [JOB 1012] sstable deleted 001604
        // INFO: [JOB 1008] compacting: sstable created 002090
        // INFO: [JOB 1010] sstable deleted 002043
        db.DeleteRange("febhkla@5", "wudlsai@17") // <nil> #7894
        db.DeleteRange("craaykk@9", "mxdmst@1") // <nil> #7895
        // INFO: [JOB 1012] sstable deleted 002080
        db.Delete("uafhw@17") // <nil> #7896
        db.Merge("mlnjzukqpek@14", "kwvizre") // <nil> #7897
        db.Set("ujatzkx", "tduub") // <nil> #7898
        snap54 = db.NewSnapshot() #7899
        db.Merge("wbnsdxaavw@11", "opwx") // <nil> #7900
        // INFO: [JOB 1013] compacting: sstable created 002091
        // INFO: [JOB 1008] compacting: sstable created 002092
        db.Delete("hwngsdr@5") // <nil> #7901
        // INFO: [JOB 1013] compacted(default) L4 [001891] (1.0 K) + L5 [] (0 B) -> L5 [002091] (1017 B), in 0.0s (0.0s total), output rate 5.0 M/s
        db.Set("grrbo@18", "") // <nil> #7902
        iter74 = db.NewIter("", "dikp@16", 2 /* key types */, 0, 0, "@1" /* masking suffix */) // <nil> #7903
        db.Set("sddrbkgaury@18", "hpiljvnkzqito") // <nil> #7904
        db.Delete("gidicrpjsx@14") // <nil> #7905
        // INFO: [JOB 1013] sstable deleted 001891
        // INFO: [JOB 1015] compacting(default) L4 [002067] (1.1 K) + L5 [001831] (1.3 K)
        // INFO: [JOB 1015] compacting: sstable created 002093
        iter74.First() // [true,"agqmmkdup@15","kzyxhqsirzvwpuidw",<no range>] <nil> #7906
        // INFO: [JOB 1008] compacted(default) L0 [001913 002019 002020 002018 002063] (5.5 K) + L4 [002087 001987 002035] (4.0 K) -> L4 [002090 002092] (3.4 K), in 0.0s (0.0s total), output rate 538 K/s
        batch53 = db.NewIndexedBatch() #7907
        iter74.SetBounds("oiwx", "pkljpuesc@12") // <nil> #7908
        iter74.SeekLT("pkljpuesc@12", "") // [true,"oiwx",<no point>,["oiwx","pkljpuesc@12")=>{"@15"="scezejiehdm","@3"="he"}*] <nil> #7909
        iter74.Next("") // [false] <nil> #7910
        iter74.Next("") // [false] <nil> #7911
        iter74.Prev("") // [true,"oiwx",<no point>,["oiwx","pkljpuesc@12")=>{"@15"="scezejiehdm","@3"="he"}*] <nil> #7912
        iter74.SeekPrefixGE("oogvf@5") // [true,"oogvf@5",<no point>,["oogvf","oogvf\x00")=>{"@15"="scezejiehdm","@3"="he"}*] <nil> #7913
        iter74.NextPrefix() // [false] <nil> #7914
        batch53.Delete("leaximgwi@5") // <nil> #7915
        // INFO: [JOB 1017] compacting(default) L0 [001342 002034 002088] (2.4 K) + L4 [] (0 B)
        --- FAIL: TestMeta/execution/random-011 (2.68s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1673069160394775357 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8230863?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8230863?buildTab=artifacts#meta) on refs/heads/master @ [65ff304d3d7a](https://github.com/cockroachdb/pebble/commits/65ff304d3d7a):


```
        // INFO: [JOB 1] sstable deleted 001786
        // INFO: [JOB 1] sstable deleted 001797
        // INFO: [JOB 1] sstable deleted 001798
        // INFO: [JOB 1] sstable deleted 001799
        // INFO: [JOB 1] sstable deleted 001800
        // INFO: [JOB 1] sstable deleted 001801
        // INFO: [JOB 1] sstable deleted 001802
        // INFO: [JOB 1] sstable deleted 001803
        // INFO: [JOB 1] sstable deleted 001804
        // INFO: [JOB 1] sstable deleted 001805
        // INFO: [JOB 1] sstable deleted 001806
        // INFO: [JOB 1] sstable deleted 001807
        // INFO: [JOB 1] sstable deleted 001819
        // INFO: [JOB 1] sstable deleted 001823
        // INFO: [JOB 1] sstable deleted 001829
        // INFO: [JOB 1] sstable deleted 001845
        // INFO: [JOB 1] sstable deleted 001846
        // INFO: [JOB 1] sstable deleted 001847
        // INFO: [JOB 1] MANIFEST deleted 001830
        db.SingleDelete("llsfg@14", false /* maybeReplaceDelete */) // <nil> #6139
        // INFO: [JOB 2] compacting: sstable created 001882
        // INFO: [JOB 2] compacting: sstable created 001883
        // INFO: [JOB 2] compacting: sstable created 001884
        // INFO: [JOB 2] compacting: sstable created 001885
        // INFO: [JOB 2] compacting: sstable created 001886
        // INFO: [JOB 4] compacting: sstable created 001887
        // INFO: [JOB 4] compacting: sstable created 001888
        // INFO: [JOB 2] compacting: sstable created 001889
        db.Get("ygjrd@14") // ["nbzuw"] <nil> #6140
        db.Merge("ktzogbaph@15", "j") // <nil> #6141
        batch46 = db.NewIndexedBatch() #6142
        iter64 = db.NewIter("", "", 2 /* key types */, 0, 0, "" /* masking suffix */) // <nil> #6144
        // INFO: [JOB 2] compacted(default) L0 [001809 001877] (1.9 K) + L5 [001787 001788 001813 001814] (3.8 K) -> L5 [001882 001883 001884 001885 001886 001889] (5.6 K), in 0.0s (0.0s total), output rate 3.0 M/s
        batch46.Delete("tcpwmhalls@6") // <nil> #6143
        batch46.Set("vruchkr@12", "kfk") // <nil> #6150
        batch46.Get("tlhzxy@3") // [""] pebble: not found #6152
        batch46.Set("svrqmcek@15", "cpbxzbsleesuwgvsrck") // <nil> #6155
        // INFO: [JOB 4] compacting: sstable created 001890
        iter64.Prev("hhbqhfu@6") // [valid,"zwneq@5","fkdqzf",<no range>] <nil> #6145
        iter64.First() // [true,"acfhggoewiz","xcmpvaymyuosdsdxi",<no range>] <nil> #6146
        iter64.First() // [true,"acfhggoewiz","xcmpvaymyuosdsdxi",<no range>] <nil> #6147
        iter64.Next("") // [true,"adhoy",<no point>,["adhoy","adotfeoay")=>{"@7"="kvswxyqbu"}*] <nil> #6148
        iter64.SeekLT("vmafyyiogzq@6", "") // [true,"uqohabxqx@1","vvdyo",["tpheehby","vjasrtgtq")=>{"@9"="cmq"}*] <nil> #6149
        iter64.Prev("") // [true,"tpheehby",<no point>,["tpheehby","vjasrtgtq")=>{"@9"="cmq"}] <nil> #6151
        batch47 = db.NewBatch() #6153
        iter64.First() // [true,"acfhggoewiz","xcmpvaymyuosdsdxi",<no range>*] <nil> #6154
        batch47.Merge("xdaamjubu@11", "qwmzdhfafibudpz") // <nil> #6156
        batch47.Merge("tbwgteszy@6", "nkgsrnoxlbtki") // <nil> #6160
        // INFO: [JOB 5] compacting(default) L0 [001594 001878] (1.5 K) + L5 [] (0 B)
        --- FAIL: TestMeta/execution/random-024 (2.63s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1673156467118646211 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8240099?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/8240099?buildTab=artifacts#meta) on refs/heads/master @ [65ff304d3d7a](https://github.com/cockroachdb/pebble/commits/65ff304d3d7a):


```
        db.SingleDelete("yfsxxv@13", false /* maybeReplaceDelete */) // <nil> #3461
        snap22.Get("dtgul@5") // ["bdwkhquxhpbqidq"] <nil> #3465
        iter32.Prev("upypaos@6") // [invalid] pebble: unsupported reverse prefix iteration #3476
        // INFO: [JOB 464] compacting: sstable created 000844
        // INFO: [JOB 466] validated table: 000829:[gfxnd@6#1029,2-wzhrerhzlod@10#1029,0]
        // INFO: [JOB 464] compacting: sstable created 000846
        // INFO: [JOB 458] compacting: sstable created 000845
        // INFO: [JOB 468] compacting(default) L4 [000594] (942 B) + L5 [] (0 B)
        iter32.First() // [true,"yzulpkzmy@6",<no point>,["yzulpkzmy@6","zfsbb")=>{"@11"="umquf"}*] <nil> #3490
        iter32.SetBounds("", "ueljmg@7") // <nil> #3499
        iter32.Next("") // [false] <nil> #3518
        db.Delete("dvdrrzgdjbe@8") // <nil> #3466
        db.Set("rllltlmhyy@4", "yhcrhlfjeainc") // <nil> #3467
        db.Delete("lzec@12") // <nil> #3468
        // INFO: [JOB 468] compacting: sstable created 000847
        // INFO: [JOB 458] compacting: sstable created 000848
        // INFO: [JOB 464] compacted(default) L3 [000712] (1.1 K) + L4 [000599] (959 B) -> L4 [000844 000846] (2.1 K), in 0.0s (0.0s total), output rate 380 K/s
        db.Get("nmkbnjo@4") // [""] pebble: not found #3469
        iter32.First() // [true,"akcrpcuax@13","rytjhmxecoqwnwri",<no range>] <nil> #3539
        iter32.SeekGE("twlotr@1", "") // [true,"twlotr@1",<no point>,["pehnfuwxte","ueljmg@7")=>{"@11"="umquf"}*] <nil> #3542
        iter32.SeekLT("bnsrxcdsqp@11", "") // [true,"bardwgvasbv",<no point>,["bardwgvasbv","bnbjoh")=>{"@11"="vtktqrx","@6"="sgalalnifruwfc"}*] <nil> #3578
        iter32.SeekGE("gkdenpkybhb@9", "") // [true,"gkdenpkybhb@9",<no point>,["gfxnd","hbubgw")=>{"@8"="fktoeekfpz","@7"="eas","@6"="ui"}*] <nil> #3587
        iter32.SeekGE("ctewj@9", "") // [true,"ctewj@9",<no point>,["ctewj","evcze")=>{"@11"="vtktqrx"}*] <nil> #3596
        iter32.SeekGE("uepyzqmej@5", "") // [false] <nil> #3609
        iter32.First() // [true,"akcrpcuax@13","rytjhmxecoqwnwri",<no range>] <nil> #3612
        iter32.Next("") // [true,"anxfho@9","j",<no range>] <nil> #3614
        iter32.SeekPrefixGE("gvgh@11") // [true,"gvgh@11","eapmx",["gvgh","gvgh\x00")=>{"@8"="fktoeekfpz","@7"="eas","@6"="ui"}*] <nil> #3615
        iter32.SeekPrefixGE("dvws@11") // [true,"dvws@11","b",["dvws","dvws\x00")=>{"@11"="vtktqrx"}*] <nil> #3617
        // INFO: [JOB 469] compacting(default) L4 [000748] (1012 B) + L5 [] (0 B)
        // INFO: [JOB 469] compacting: sstable created 000849
        // INFO: [JOB 469] compacted(default) L4 [000748] (1012 B) + L5 [] (0 B) -> L5 [000849] (1012 B), in 0.0s (0.0s total), output rate 4.7 M/s
        // INFO: [JOB 458] compacting: sstable created 000850
        // INFO: [JOB 458] compacting: sstable created 000851
        // INFO: [JOB 468] compacted(default) L4 [000594] (942 B) + L5 [] (0 B) -> L5 [000847] (942 B), in 0.0s (0.0s total), output rate 53 K/s
        db.Set("azkfg@11", "cxskhuwbmlccuy") // <nil> #3474
        // INFO: [JOB 458] compacted(default) L0 [000618 000616 000620 000619 000615 000621 000617 000623 000627 000628 000629 000657 000659 000658 000661 000666 000663 000667 000669 000670 000672 000673 000676 000677 000678 000679 000680 000681 000682 000665 000683 000662 000664 000668 000660 000731 000733 000734 000742 000745 000746 000747 000749 000750 000752 000753 000737 000781] (49 K) + L3 [000705 000771 000772 000775 000738] (6.1 K) -> L3 [000800 000801 000802 000803 000804 000805 000806 000807 000808 000809 000810 000811 000812 000813 000814 000815 000816 000817 000818 000828 000843 000845 000848 000850 000851] (31 K), in 0.1s (0.1s total), output rate 414 K/s
        db.Set("wzhrerhzlod@10", "fkgyltt") // <nil> #3477
        iter30.Next("") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #3479
        iter30.SeekPrefixGE("cnad@3") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #3480
        iter30.Prev("") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #3485
        // INFO: [JOB 472] compacting(default) L4 [000751] (1007 B) + L5 [] (0 B)
        // INFO: [JOB 473] compacting(default) L4 [000846] (1.0 K) + L5 [000645] (951 B)
        snap25 = db.NewSnapshot() #3487
        // INFO: [JOB 472] compacting: sstable created 000852
        // INFO: [JOB 474] compacting(move) L3 [000801] (1.0 K) + L4 [] (0 B)
        // INFO: [JOB 474] compacted(move) L3 [000801] (1.0 K) + L4 [] (0 B) -> L4 [000801] (1.0 K), in 0.0s (0.0s total), output rate 219 M/s
        db.Merge("yhhguojwsu@13", "uhalyiwxyqfxnqxdoqg") // <nil> #3488
        db.Delete("bnykzn") // <nil> #3491
        // INFO: [JOB 458] sstable deleted 000771
        --- FAIL: TestMeta/execution/random-003 (1.45s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1673243746674806693 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/range_keys.go
+++ b/range_keys.go
@@ -501,24 +501,6 @@ func (i *lazyCombinedIter) initCombinedIteration(
 		}
 	}
 
-	if i.parent.hasPrefix {
-		si := i.parent.comparer.Split(seekKey)
-		if i.parent.cmp(seekKey[:si], i.parent.prefixOrFullSeekKey) > 0 {
-			// The earliest possible range key has a start key with a prefix
-			// greater than the current iteration prefix. There's no need to
-			// switch to combined iteration, because there are not any range
-			// keys within the bounds of the prefix. Additionally, using a seek
-			// key that is outside the scope of the prefix can violate
-			// invariants within the range key iterator stack. Optimizations
-			// that exit early due to exhausting the prefix may result in
-			// `seekKey` being larger than the next range key's start key.
-			//
-			// See the testdata/rangekeys test case associated with #1893.
-			i.combinedIterState = combinedIterState{initialized: false}
-			return pointKey, pointValue
-		}
-	}
-
 	// An operation on the point iterator observed a file containing range keys,
 	// so we must switch to combined interleaving iteration. First, construct
 	// the range key iterator stack. It must not exist, otherwise we'd already

--- a/testdata/iter_histories/lazy_combined_iteration
+++ b/testdata/iter_histories/lazy_combined_iteration
@@ -220,3 +220,60 @@ a: (a, .)
 using lazy iterator
 n: (., [m-z) @5=foo UPDATED)
 using combined (non-lazy) iterator
+
+# Regression tests for #2210 metamorphic test failure.
+#
+# Lazy-combined iteration depends on individual point level iterators triggering
+# a switch to combined iteration when they observe a file containing relevant
+# range keys. Previously, this switch did not happen if the observed range
+# keys all lied outside the current iteration prefix.
+#
+# This made it possible for a level to become positioned beyond the file
+# containing range keys, without ever triggering the switch to combined
+# iteration. A subsequent seek that made use of the TrySeekUsingNext
+# optimization would never observe the file containing range keys, and omit the
+# range keys.
+
+define
+L6
+  bax.DEL.9:
+L6
+  rangekey:c-d:{(#0,RANGEKEYSET,@1,foo)}
+L6
+  d@2.SET.2:v
+----
+6:
+  000004:[bax#9,DEL-bax#9,DEL]
+  000005:[c#0,RANGEKEYSET-d#72057594037927935,RANGEKEYSET]
+  000006:[d@2#2,SET-d@2#2,SET]
+
+combined-iter
+seek-prefix-ge bax
+seek-prefix-ge cat
+----
+.
+cat: (., [cat-"cat\x00") @1=foo UPDATED)
+
+# Another regression test for the #2210 metamorphic test failure, this one using
+# a MERGE key to force the Iterator to step the internal iterator beyond the
+# range key file.
+
+define
+L6
+  bax.MERGE.9:v
+L6
+  rangekey:c-d:{(#0,RANGEKEYSET,@1,foo)}
+L6
+  d@2.SET.2:v
+----
+6:
+  000004:[bax#9,MERGE-bax#9,MERGE]
+  000005:[c#0,RANGEKEYSET-d#72057594037927935,RANGEKEYSET]
+  000006:[d@2#2,SET-d@2#2,SET]
+
+combined-iter
+seek-prefix-ge bax
+seek-prefix-ge cat
+----
+bax: (v, .)
+cat: (., [cat-"cat\x00") @1=foo UPDATED)

